frontend/src/metabase/dashboard/reducers.js

```diff
@@ -278,7 +278,13 @@ const parameterValues = handleActions(
   {
     [INITIALIZE]: { next: () => ({}) }, // reset values
     [SET_PARAMETER_VALUE]: {
-      next: (state, { payload: { id, value } }) => assoc(state, id, value),
+      next: (state, { payload: { id, value, isDraft } }) => {
+        if (!isDraft) {
```

Comment:
as mentioned earlier `SET_PARAMETER_VALUE` will have a new payload attribute `isDraft` which determines which state we should set the parameter value to.